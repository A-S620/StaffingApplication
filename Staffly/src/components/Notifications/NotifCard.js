import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import {Card, CardContent, Grid, Typography} from '@material-ui/core'
import ClearIcon from '@material-ui/icons/Clear';
import {connect} from "react-redux";
import IconButton from "@material-ui/core/IconButton";
import {store} from "../../store/Store";

const useStyles = makeStyles({
    root: {
        minWidth: 275,
        margin: "10px",
    },
    bullet: {
        display: 'inline-block',
        margin: '0 2px',
        transform: 'scale(0.8)',
    },
    title: {
        fontSize: 14,
    },
    pos: {
        marginBottom: 12,
    },

});

function NotifCard(props) {
    const classes = useStyles();
    const bull = <span className={classes.bullet}>â€¢</span>;
    const [cardData, setCardData] = React.useState(props.notifs);
    // setCardData(props);
    // console.log(cardData[0].NotifId);

    // {props.notifs.map((singleNotif) => {
    //
    //     return (
    //         cardData.push(singleNotif)
    //     )
    //
    // })}
    const handleDelete = (cardToDelete) => () => {
        console.log(cardToDelete.NotifId);
        var index= cardData.indexOf(cardToDelete);
        console.log(index);
        cardData.splice(index,1);
        // setCardData((cards) => cards.filter((card) => card.NotifId !== cardToDelete.NotifId));
    };
    return (

        <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
        >

            {props.notifs.map((data) =>{
                return(
                    <Card className={classes.root}>
                        <Grid
                            container
                            direction="row"
                            justify="flex-end"
                            alignItems="flex-start"
                        >
                            <IconButton
                                onClick={ () =>{
                                    store.dispatch({
                                        type: "DELETE_NOTIFICATION",
                                        payload: data.NotifId
                                    })
                                }

                                    }
                            >
                                <ClearIcon/>
                            </IconButton>

                        </Grid>
                        <CardContent>
                            <Typography variant="h5" component="h2">
                                {data.NotifTitle}
                            </Typography>
                            <Typography className={classes.pos} color="textSecondary">
                                {data.NotifDate}
                            </Typography>
                            <Typography variant="body2" component="p">
                                {data.NotifInfo}
                            </Typography>
                        </CardContent>

                    </Card>
                )
            })}




            {/*<Card className={classes.root}>*/}
            {/*    <CardContent>*/}
            {/*        <Typography className={classes.title} color="textSecondary" gutterBottom>*/}
            {/*            Word of the Day*/}
            {/*        </Typography>*/}
            {/*        <Typography variant="h5" component="h2">*/}
            {/*            be{bull}nev{bull}o{bull}lent*/}
            {/*        </Typography>*/}
            {/*        <Typography className={classes.pos} color="textSecondary">*/}
            {/*            adjective*/}
            {/*        </Typography>*/}
            {/*        <Typography variant="body2" component="p">*/}
            {/*            well meaning and kindly.*/}
            {/*            <br/>*/}
            {/*            {'"a benevolent smile"'}*/}
            {/*        </Typography>*/}
            {/*    </CardContent>*/}
            {/*    <CardActions>*/}
            {/*        <Button size="small">Learn More</Button>*/}
            {/*    </CardActions>*/}
            {/*</Card>*/}
        </Grid>
    );
}

const mapStateToProps = state => {
    return {
        notifs: state.notifications
    }
}
export default connect(mapStateToProps, {})(NotifCard);
