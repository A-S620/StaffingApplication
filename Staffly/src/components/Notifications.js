import React, { Component } from 'react';
import NotifCard from "./Notifications/NotifCard";


class Notifications extends Component {
    render() {
        return (
            <div className="Notifications" style={{marginTop:60}} >
                <NotifCard/>
            </div>
        );
    }
}

export default Notifications;