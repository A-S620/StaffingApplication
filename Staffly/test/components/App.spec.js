import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux'
import Notifications from "../../src/components/Notifications";
describe('MyComponent', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Notifications debug />);

        expect(component).toMatchSnapshot();
    });
});
describe('test jest', () => {
    it('should work', () => {
        expect(5).toBe(5);
    });
});