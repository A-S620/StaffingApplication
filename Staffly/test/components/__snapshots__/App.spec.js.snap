// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="Notifications"
  style={
    Object {
      "marginTop": 60,
    }
  }
>
  <div>
    test 5
  </div>
</div>
`;
